docs/basic_construction.md

```diff
@@ -0,0 +1,125 @@
+This construction uses a recursive filtering process, guided by random oracles, to produce succinct and efficient proofs.
+
+## Overview
+- The prover builds a proof by creating tuples (sequences) of elements from $S_p$, the set of elements. 
+- The process begins with roots as starting points and expands them step-by-step. 
+- At each step, only tuples that satisfy specific conditions, as determined by a random oracle, are kept for further extension.
+- The random oracle acts as a filter to ensure that the prover selects only a small subset of elements that is highly unlikely to satisfy the conditions by chance alone.
+- The proof construction uses *Depth-First Search (DFS)* over a conceptual tree, where each path represents a potential tuple. 
+- The DFS ensures efficiency by pruning invalid branches early.
+
+### Core Components
+- **Parameters**:
+  - $d: ~~$ Maximum number of subtrees to search to find a proof.
+  - $u: ~~$ The length of the proof tuple generated by the prover.
+  - $q: ~~$ The probability that a tuple of full size is selected.
+  - $n_p: ~~$ Total number of elements available to the prover.
+  - $n_f: ~~$ The lower bound to prove on prover set.
+
+- **Random Oracles**:
+  - $H_1: ~~$ A random oracle returning $1$ with probability $1/n_p$, used to probabilistically filter candidate tuples. 
+  - $H_2: ~~$ Another random oracle returning $1$ with probability $q$, applied as a final test to determine if a tuple qualifies as a valid proof.
+
+### The Tree Structure
+- Each root $t \in \[d\]$ is treated as the starting node of a tree.
+- The tree has $u$ levels, where $u$ is the length of the proof tuple.
+- At level $i$, nodes represent tuples of the form $(t, s_1, s_2, \ldots, s_i)$, where $t$ is the root, and $s_1, \ldots, s_i$ are elements.
+
+### Tuple Construction via DFS
+- The prover constructs tuples recursively using DFS.
+- At each level $i$, the prover:
+  - Extends the current tuple by adding an element $s_{i+1}$ from $S_p$.
+  - Checks tuple validity using the random oracle $H_1$. A tuple $(t, s_1, \ldots, s_{i+1})$ is valid if $H_1(t, s_1, \ldots, s_{i+1}) = 1$.
+  - If the tuple fails the $H_1$ check, it is discarded, and the DFS does not explore its descendants.
+- The process continues until a valid tuple of length $u$ is found, or all possibilities are exhausted.
+
+#### Example: how DFS works
+- Choose a root $t \in \[d\]$ and initialize the search with $(t)$.
+  - At each step, extend the current tuple $(t, s_1, \ldots, s_i)$ by trying every element $s_{i+1} \in S_p$.
+  - Check if $H_1(t, s_1, \ldots, s_{i+1}) = 1$:
+    - If true, recursively extend the tuple.
+    - If false, backtrack and try the next candidate for $s_{i+1}$.
+- Stop when a valid tuple $(t, s_1, \ldots, s_u)$ is found and satisfies $H_2(t, s_1, \ldots, s_u) = 1$.
+
+## The Protocol
```

Comment:
Should you add a parameter generation algorithm? I think it would be good to write that $u$ is a function of $l_sec / log(n_p/n_f)$